File: common/src/main/java/com/ssblur/scriptor/word/descriptor/discount/WeakDescriptor.java
package com.ssblur.scriptor.word.descriptor.discount;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.helpers.targetable.EntityTargetable;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.word.descriptor.CastDescriptor;
import net.minecraft.world.effect.MobEffects;
import net.minecraft.world.entity.LivingEntity;

public class WeakDescriptor extends Descriptor implements CastDescriptor {
  @Override
  public Cost cost() {
    return new Cost(0.9d, COSTTYPE.MULTIPLICATIVE);
  }

  @Override
  public boolean cannotCast(Targetable caster) {
    if(caster instanceof EntityTargetable entityTargetable && entityTargetable.getTargetEntity() instanceof LivingEntity living)
      return !living.hasEffect(MobEffects.WEAKNESS);
    return true;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/discount/WeakDescriptor.kt
package com.ssblur.scriptor.word.descriptor.discount

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.EntityTargetable
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.CastDescriptor
import net.minecraft.world.effect.MobEffects
import net.minecraft.world.entity.LivingEntity

class WeakDescriptor : Descriptor(), CastDescriptor {
    override fun cost(): Cost {
        return Cost(0.9, COSTTYPE.MULTIPLICATIVE)
    }

    override fun cannotCast(caster: Targetable?): Boolean {
        if (caster is EntityTargetable && caster.targetEntity is LivingEntity) return !(caster.targetEntity as LivingEntity).hasEffect(MobEffects.WEAKNESS)
        return true
    }
}
